### PR TITLE
config: move storage check after initializeTempDir

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?
Move storage check after initializeTempDir so that avoiding the capacity we get is smaller than the fact.

### Related changes

- Need to cherry-pick to the release branch

### Release note 
None

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -864,20 +863,6 @@ func (c *Config) Valid() error {
...
-	if c.OOMUseTmpStorage {
-		if c.TempStorageQuota < 0 {
-			// means unlimited, do nothing
-		} else {
-			capacityByte, err := storage.GetTargetDirectoryCapacity(c.TempStoragePath)
-			if err != nil {
-				return err
-			} else if capacityByte > uint64(c.TempStorageQuota) {
-				return fmt.Errorf("value of [temp-storage-quota](%d byte) exceeds the capacity(%d byte) of the [%s] directory", c.TempStorageQuota, capacityByte, c.TempStoragePath)
-			}
```
Comment (Yisaer) [inline config/config.go:877]: The original logic here is wrong, I correct it in this request.

**Comment by XuHuaiyu** [comment]
@Yisaer Please cherry-pick this commit to release-4.0

**Pull request merged**